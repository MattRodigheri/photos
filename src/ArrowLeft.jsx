import React from 'react';
import arrows from './../public/arrows.css';


const ArrowLeft = (props) => {
  return <img className='arrow left-arrow' src='https://s3-us-west-1.amazonaws.com/indistinctshoutingphotos/left-arrow.gif' onClick={props.previousPhoto}/>
}

export default ArrowLeft;
