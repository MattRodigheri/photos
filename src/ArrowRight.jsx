import React from 'react';

class ArrowRight extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.nextPhoto();
  }


  render() {
    return(
      <img className='arrow right-arrow' src='../right-arrow.png' onClick={this.handleClick} />
    )
  }
}

export default ArrowRight;
