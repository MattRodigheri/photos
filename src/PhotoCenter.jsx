import React from 'react';

const PhotoCenter = (props) => {
  return <img images={props.images} src={`${props.images[1]}`} />
}

export default PhotoCenter;
