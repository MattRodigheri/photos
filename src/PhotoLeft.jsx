import React from 'react';

const PhotoLeft = (props) => {
  return <img images={props.images} src={`${props.images[0]}`} />
}

export default PhotoLeft;
