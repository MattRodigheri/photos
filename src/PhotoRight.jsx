import React from 'react';
import ArrowRight from './ArrowRight.jsx';


var nextPhoto = function() {
  console.log('next')
}

const PhotoRight = (props) => {
  return (
    <div>
      <ArrowRight />
      <img className='photo' images={props.images} src={`${props.images[2]}`} />
    </div>
  )
}

export default PhotoRight;
