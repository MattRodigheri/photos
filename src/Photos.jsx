import React from 'react';
import PhotoLeft from './PhotoLeft.jsx';
import PhotoCenter from './PhotoCenter.jsx';
import PhotoRight from './PhotoRight.jsx';
import ArrowLeft from './ArrowLeft.jsx';
import ArrowRight from './ArrowRight.jsx';

class Photos extends React.Component {
  constructor(props) {
    super(props);

  }

  previousPhoto() {

  }

  nextPhoto() {

  }

  render() {
    return (
      <div>
        <PhotoLeft images={this.props.images} />
        <PhotoCenter images={this.props.images} />
        <PhotoRight images={this.props.images} />

        <ArrowLeft previousPhoto={this.previousPhoto}/>
        <ArrowRight nextPhoto={this.nextPhoto}/>
      </div>
    )
  }
}

export default Photos;
