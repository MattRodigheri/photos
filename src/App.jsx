import React from 'react';
import $ from 'jquery';
import Photos from './Photos.jsx';

class App extends React.Component {
  constructor() {
    super();

    this.state = {
      images: [],
      photographers: [],
      friends: [],
      reviews: []
    }

    this.getData = this.getData.bind(this);
    this.nextPhoto = this.nextPhoto.bind(this);
    this.previousPhoto = this.previousPhoto.bind(this);
    this.timedScroll = window.setInterval(this.nextPhoto, 8000);
  }

  componentDidMount() {
    this.getData();
  }

  getData() {
    var context = this;
    var tenImages = [];
    var tenPhotographers = [];
    var tenFriends = [];
    var tenReviews = [];
    function getRandomInt(min, max) {
      return Math.floor(Math.random() * (Math.floor(max) - Math.ceil(min) + 1)) + Math.ceil(min);
    }
    $.ajax({
      url: '/photos/',
      method: 'GET',
      success: function(data) {
        var count = 0;
        while (count < 10) {
          var randomFoodPic = getRandomInt(0, 49);
          tenImages.push(data[randomFoodPic].url);
          tenPhotographers.push(data[randomFoodPic].photographer);
          tenFriends.push(data[randomFoodPic].friends);
          tenReviews.push(data[randomFoodPic].reviews);
          count++;
        }
        context.setState({
          images: tenImages,
          photographers: tenPhotographers,
          friends: tenFriends,
          reviews: tenReviews
        })
      },
      error: function(err) {
        console.log(err)
      }
    });
  }

  nextPhoto() {
    var newArr = this.state.images;
    var hold = newArr.shift();
    newArr.push(hold);
    this.setState({
      images: newArr
    })
  }

  previousPhoto() {
    var newArr = this.state.images;
    var hold = newArr.pop();
    newArr.unshift(hold);
    this.setState({
      images: newArr
    })
  }


  render() {
    return (
      <div>
        <Photos info={this.state} nextPhoto={this.nextPhoto} previousPhoto={this.previousPhoto}/>
      </div>
    )
  }
}

export default App;
