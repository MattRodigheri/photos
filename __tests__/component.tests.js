import React from 'react';
import App from '../src/App.jsx';
import Photos from '../src/Photos.jsx';
import PhotoLeft from '../src/PhotoLeft.jsx'
import PhotoCenter from '../src/PhotoCenter.jsx'
import PhotoRight from '../src/PhotoRight.jsx'
import ArrowLeft from '../src/ArrowLeft.jsx';
import ArrowRight from '../src/ArrowRight.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';


Enzyme.configure({ adapter: new Adapter() });


describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('checks the state', async () => {
    const wrapper = shallow(<App />);
    expect(Array.isArray(wrapper.state('images'))).toBe(true);
  })
});



describe('Photos', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Photos />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('PhotoLeft', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<img className='photo left-photo'/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('PhotoCenter', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<img className='photo center-photo'/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('PhotoRight', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<img className='photo right-photo'/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ArrowLeft', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ArrowLeft />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ArrowRight', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ArrowRight />);
    expect(wrapper).toMatchSnapshot();
  });
});
