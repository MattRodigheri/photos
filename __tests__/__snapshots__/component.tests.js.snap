// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Photos
        images={Array []}
        nextPhoto={[Function]}
        previousPhoto={[Function]}
      />,
      "className": "slideshow",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "images": Array [],
        "nextPhoto": [Function],
        "previousPhoto": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Photos
          images={Array []}
          nextPhoto={[Function]}
          previousPhoto={[Function]}
        />,
        "className": "slideshow",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "images": Array [],
          "nextPhoto": [Function],
          "previousPhoto": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ArrowLeft renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArrowLeft />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "arrow left-arrow",
      "onClick": [Function],
      "src": "../left-arrow.png",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "arrow left-arrow",
        "onClick": [Function],
        "src": "../left-arrow.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ArrowRight renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArrowRight />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "arrow right-arrow",
      "onClick": [Function],
      "src": "../right-arrow.png",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "arrow right-arrow",
        "onClick": [Function],
        "src": "../right-arrow.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PhotoCenter renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <img
    className="photo center-photo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "photo center-photo",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "photo center-photo",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PhotoLeft renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <img
    className="photo left-photo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "photo left-photo",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "photo left-photo",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`PhotoRight renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <img
    className="photo right-photo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "photo right-photo",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "photo right-photo",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Photos renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Photos />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PhotoLeft />,
        <PhotoCenter />,
        <PhotoRight />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "images": undefined,
          "previousPhoto": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "images": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "images": undefined,
          "nextPhoto": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PhotoLeft />,
          <PhotoCenter />,
          <PhotoRight />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "images": undefined,
            "previousPhoto": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "images": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "images": undefined,
            "nextPhoto": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
